amplicon_stats.c

```diff
@@ -928,7 +932,7 @@ int dump_stats(char type, char *name, astats_t *stats, astats_args_t *args,
             double n1 = stats->nbases[i] / alen;
             double v = stats->nbases2[i] / (alen*alen) /nfile
                 - (n1/nfile)*(n1/nfile);
-            fprintf(ofp, "\t%.1f", sqrt(v));
+            fprintf(ofp, "\t%.1f", sqrt(v+1e-10));
```

Comment:
This didn't work for my test file [MILK-10285F1.210210_A01250_0009_AH32GCDRXY.2t316.cram](ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR531/ERR5313570/MILK-10285F1.210210_A01250_0009_AH32GCDRXY.2t316.cram) as I still got some `nan`s.  Possibly better might be to multiply through by `nfile` and pull the denominator (now all squared) out of the `sqrt()` resulting in:
```
   double alen = amp[i].min_right - amp[i].max_left+1;
   fprintf(ofp, "\t%.1f", sqrt(stats->nbases2[i] * nfile - stats->nbases[i] * stats->nbases[i]) / alen / nfile);
```
However, some of the values in `stats->nbases2` can get a bit big, for example in the test file above:
```
p stats->nbases2[6]
$1 = 226819353033529
```
This needs 48 bits, which doesn't leave that much head-room before it overflows.  It might be a good idea to look into [Welford's algorithm](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm) to see if it can calculate the variance without the accumulated numbers getting too big.

Reply:
Maybe 1e-10 is too small then, but we could just add a `nfile>1?sqrt(v):0` instead as that's realistically the only time this matters.

I'm not so worried about sizes yet.  For nbases^2 to break 64-bit we'd need 32-bits worth of combined depth across the samples.  I don't think we're going to be more than 1 billion fold coverage yet!  We don't need extra complication here until it really matters, and I don't think it does.  Note we're printing this up with `%.1f` so accuracy isn't vital.